store/src/java/com/zimbra/cs/ml/classifier/ClassifierData.java

```diff
@@ -0,0 +1,54 @@
+package com.zimbra.cs.ml.classifier;
+
+import com.zimbra.cs.mailbox.Message;
+import com.zimbra.cs.ml.Classifiable;
+import com.zimbra.cs.ml.classifier.Classifier.ClassifiableType;
+import com.zimbra.cs.ml.feature.FeatureSet;
+import com.zimbra.cs.ml.schema.ClassifierInfo;
+import com.zimbra.cs.ml.schema.ClassifierSpec;
+
+/**
+ * Class encompassing all data needed to register a classifier
+ */
+public class ClassifierData<T extends Classifiable> {
+    private String label;
+    private String description;
+    private FeatureSet<T> featureSet;
+    private ClassifierSpec spec;
+    private ClassifiableType type;
+
+    public ClassifierData(ClassifiableType type, String label, ClassifierSpec spec, FeatureSet<T> featureSet) {
+        this(type, label, spec, featureSet, null);
+    }
+
+    public ClassifierData(ClassifiableType type, String label, ClassifierSpec spec, FeatureSet<T> featureSet, String description) {
+        this.type = type;
+        this.label = label;
+        this.description = description;
+        this.featureSet = featureSet;
+        this.spec = spec;
+    }
+
+    public String getLabel() {
+        return label;
+    }
+
+    public ClassifierSpec getSpec() {
+        return spec;
+    }
+
+    @SuppressWarnings("unchecked")
+    public Classifier<T> create(ClassifierInfo info) {
+        Classifier<T> classifier;
+        switch (type) {
```

Comment:
Kind of looks like don't need the switch statement?  Or perhaps need a comment like similar cases lower down.  Would it make more sense to only allow MESSAGE at the moment and anything else is an error?